compiler/passes/normalize.cpp

```diff
@@ -2833,18 +2833,76 @@ static void fixupExportedArrayFormals(FnSymbol* fn) {
         // TODO: handle call expression stuff.
       }
 
-      // Create a representation of the array argument that is accessible
-      // outside of Chapel (in the form of a pointer and a corresponding size)
-      formal->typeExpr->replace(new BlockStmt(new SymExpr(dtExternalArray->symbol)));
+      SymExpr* dom = toSymExpr(call->get(1));
```

Comment:
Above, there is the comment
`call is chpl__buildArrayRuntimeType, which takes in a domain`

it seems you should verify this with an INT_ASSERT.